ts/Data/DataPool.ts

```diff
@@ -120,20 +149,44 @@ class DataPool implements DataEvent.Emitter {
     ): Promise<DataConnector> {
         const connector = this.connectors[name];
 
+        // already loaded
         if (connector) {
-            // already loaded
             return Promise.resolve(connector);
         }
 
+        let waiting = this.waiting[name];
+
+        // currently loading
+        if (waiting) {
+            return new Promise((resolve: ConnectorResolve): void => {
+                waiting.push(resolve);
+            });
+        }
+
+        this.waiting[name] = waiting = [];
+
         const connectorOptions = this.getConnectorOptions(name);
 
         if (connectorOptions) {
-            return this.loadConnector(connectorOptions);
+            return this
+                .loadConnector(connectorOptions)
+                .then((connector): DataConnector => {
+                    delete this.waiting[name];
+
+                    window.setTimeout((): void => {
+                        for (let i = 0, iEnd = waiting.length; i < iEnd; ++i) {
+                            waiting[i](connector);
+                        }
+                    }, 1);
```

Comment:
Do we need this 1ms here?

Reply:
Yes. :) So everything gets resolved in order.